### ISSUE TITLE
Bug: must mention the directive name when only binding to one of its outputs

### ISSUE THREAD

**Issue by wardbell** [angular/angular#5707]
Detected in  alpha 42..48.  @vsavkin thinks it might be fixed coincidentally in a.49.

Recording the report here so we can track and confirm.

## Report
The following works

```
<div my-click (my-click)="clicks=$event">click me</div>
```
but this doesn't (the directive is never instantiated)
```
<div (my-click)="clicks=$event">click me</div>
```
for directive:
```
@Directive({selector:'[my-click]'})
export class MyClickDirective {
  @Output() myClick = new EventEmitter<string>();
  constructor(el: ElementRef){
    el.nativeElement
      .addEventListener('click', (event:Event) => {
        this.myClick.emit('Click!');
      });
  }
}
```

**Comment by jamesfromparis** [comment]
Hi,

Under Plunker of angular.io (live exemple), the directive works. Once I have downloaded this code, this directive does not work anymore.

- code under Plunker : works
  click me
  Click!    <------ when i click on (click me) i've got this word

- once code downloaded : does not work
  click me
     <------ when i click on (click me) i've got anything

 Could you help me?

Thanks in advance

**Comment by jamesfromparis** [comment]
Hi,
I am sorry for the inconvenience.
After adjusting, the directive works now.
Best regards

**Comment by wardbell** [comment]
**The bug remains as of beta.0**

For repro, see `app.component.html` in [plunker](https://angular.io/resources/live-examples/template-syntax/ts/plnkr.html) from the Angular Template Syntax chapter.

Find mention of this issue (.."issues/5707"). Try the `myClick` directive written as
```
<div myClick (myClick)="clicked=$event">click me</div>
```
It displays "Clicked!"

Now rewrite as (just the event binding):
```
<div (myClick)="clicked=$event">click me</div>
```
It stops working.

**Comment by bennadel** [comment]
I was about to file this as a feature request. Glad I was able to spot yours first. 

I wrote about it here. http://www.bennadel.com/blog/3004-attribute-directive-selectors-and-outputs-cannot-have-the-same-name-in-angularjs-2-beta-1.htm

Would be really nice if this worked. The primary use-case that I can think of would be to simulate non-native browser events. Like "triple-click" or "hover-intent". Things like that.

**Comment by wardbell** [comment]
Still not fixed as of beta.2

**Comment by cexbrayat** [comment]
Looks like it was fixed by #6870

**Comment by bennadel** [comment]
Awesome pants!!!

**Comment by bennadel** [comment]
Confirmed fix: http://www.bennadel.com/blog/3024-selectors-and-outputs-can-have-the-same-name-in-angular-2-beta-6.htm

Very groovy stuff :D

**Comment by vsavkin** [comment]
Fixed via 231773ea762ad45df1612efaeeec29d9b4ebca68
